custom/enikshay/integrations/nikshay/tests/test_repeaters.py

```diff
@@ -1164,12 +1169,16 @@ def test_trigger(self):
         self.assertEqual(0, len(self.repeat_records().all()))
         person = self.create_case(self.person)[0]
         update_case(self.domain, self.person_id, {ENROLLED_IN_PRIVATE: "true"})
-        with self.assertRaisesMessage(
-                NikshayLocationNotFound,
-                "Location with id {location_id} not found. This is the owner for person with "
-                "id: {person_id}".format(location_id=person.owner_id, person_id=self.person_id)
-        ):
+        with capture_log_output('notify') as logoutput:
             self._create_nikshay_enabled_case(set_property=PRIVATE_PATIENT_EPISODE_PENDING_REGISTRATION)
+        self.assertIn(NikshayLocationNotFound.__name__, logoutput.get_output())
+        expected_message = (
+            "Location with id {location_id} not found. "
+            "This is the owner for person with id: {person_id}".format(
+                location_id=person.owner_id, person_id=self.person_id)
+        )
+        self.assertIn(expected_message, logoutput.get_output())
```

Comment:
@mkangia @proteusvacuum FYI this is changing the way errors that are raised by `case_post_save` signal handlers are treated.

Previously this would have resulted in a 500 response to the phone and it would have retried the form at a later stage but the signal would NOT have been re-fired since the second form submission would be marked as a duplicate.

With this change the phone would get a 422 response and possibly retry. If the phone retried (or the reprocess queue picked up the UnfinishedSubmissionStub) then the signals would get re-fired.

Note that we'll still get a Sentry notification from the error.

Reply:
Thanks for the notification @snopoke. Its really difficult to understand the impact of this change without knowing how form processing works and what all this PR changed. 
Can you share what would be the change in the outcome from the repeater perspective and not from the phone perspective? 
Like a repeater that runs though the `case_post_save` signal will error out when it is attempted first. And then what happens?

Reply:
This should improve things from a repeater perspective. If the signal handler errors then 2 things happen:
1. The UnfinishedSubmissionStub will get picked up by the async process to re-fire the signals
2. The response to the phone will be 422 and if the phone retires it will re-fire the signals (assuming they haven't already been re-fired by the async procesS)

Reply:
Interesting. I didn't realize that if there was an error in the signal that the form would be retried later by the phone. Errors in the signals are often to do with some data error that wouldn't be fixed by resubmitting the form. Do you think we should raise errors in a different fashion to prevent the phone from needing to do extra work? 

I don't think that re-firing the signals here is actually desired, as that will create new (potentially duplicate) RepeatRecords, which already have their own retry mechanism.

Reply:
> I don't think that re-firing the signals here is actually desired, as that will create new (potentially duplicate) RepeatRecords, which already have their own retry mechanism.

was this issue addressed?
@snopoke @proteusvacuum

Reply:
@proteusvacuum @mkangia sorry I thought I had responded to this. If an error is raised when the signal is fired then no repeat record should be created at all so the repeat record retry mechanism won't kick in.